crates/sui-indexer/src/handlers/checkpoint_handler.rs

```diff
@@ -125,15 +125,11 @@ where
             .get_checkpoint(seq.into())
             .await?;
 
-        let transactions = join_all(
-            checkpoint
-                .transactions
-                .iter()
-                .map(|tx_digest| self.rpc_client.read_api().get_transaction(*tx_digest)),
-        )
-        .await
-        .into_iter()
-        .collect::<Result<Vec<_>, _>>()?;
+        let transactions = self
+            .rpc_client
+            .read_api()
+            .multi_get_transactions(checkpoint.transactions.iter().map(|t| t.clone()).collect())
```

Comment:
nit: `into_iter` to avoid the clone?

Reply:
`checkpoint` is used later, into_iter() here would cause checkpoint partially moved.